### ISSUE TITLE
bad argument #1 to 'format' (string expected, got nil)

### ISSUE THREAD

**Issue by yangm97** [group-butler/GroupButler#318]
```
An #error occurred.
./lua/plugins/users.lua:224: bad argument #1 to 'format' (string expected, got nil)
stack traceback:
 [C]: in function 'format'
 ./lua/plugins/users.lua:224: in function <./lua/plugins/users.lua:208>
 (tail call): ?
 [C]: in function 'xpcall'
 ./lua/main.lua:107: in function <./lua/main.lua:61>
 (tail call): ?
 polling.lua:47: in main chunk
 [C]: ?
en
###cb:userbutton:remwarns:[redacted]
```

```
An #error occurred.
./lua/utilities.lua:637: bad argument #2 to 'format' (string expected, got nil)
stack traceback:
 [C]: in function 'format'
 ./lua/utilities.lua:637: in function 'getSettings'
 ./lua/plugins/dashboard.lua:109: in function <./lua/plugins/dashboard.lua:89>
 (tail call): ?
 [C]: in function 'xpcall'
 ./lua/main.lua:107: in function <./lua/main.lua:61>
 (tail call): ?
 polling.lua:47: in main chunk
 [C]: ?
en
###cb:dashboard:settings:[redacted]
```

**Comment by yangm97** [comment]
```
lua: ./lua/main.lua:144: bad argument #2 to 'format' (string expected, got table)
stack traceback:
	[C]: in function 'format'
	./lua/main.lua:144: in function <./lua/main.lua:61>
	(tail call): ?
	polling.lua:47: in main chunk
	[C]: ?
```

**Comment by yangm97** [comment]
Fixed by #323

**Comment by yangm97** [comment]
Maybe not. The issue with `###cb:dashboard:settings` persists.

**Comment by riskey95** [comment]
can you provide error log? i just pull from this repo, master branch and and launch (w/o docker) but not facing issue on ###cb:dashboard:settings

**Comment by yangm97** [comment]
Ooops, I might have forgotten to update the running bots 😅 

TODO: Add CI/CD

**Comment by yangm97** [comment]
```
An #error occurred.
./lua/utilities.lua:608: bad argument #1 to 'format' (string expected, got nil)
stack traceback:
 [C]: in function 'format'
 ./lua/utilities.lua:608: in function 'getSettings'
 ./lua/plugins/dashboard.lua:109: in function <./lua/plugins/dashboard.lua:89>
 (tail call): ?
 [C]: in function 'xpcall'
 ./lua/main.lua:112: in function <./lua/main.lua:61>
 (tail call): ?
 polling.lua:47: in main chunk
 [C]: ?
it_IT
###cb:dashboard:settings:
```

Nah, seems like I did not forget to update it after all.

**Comment by riskey95** [comment]
it seem you forget to update after all, i just test it using italian but i didnt face this issue

**Comment by yangm97** [comment]
It is happening on official GB and I can assure it is up to date.

**Comment by yangm97** [comment]
```
n #error occurred.
./lua/utilities.lua:594: bad argument #1 to 'format' (string expected, got nil)
stack traceback:
 [C]: in function 'format'
 ./lua/utilities.lua:594: in function 'getSettings'
 ./lua/plugins/dashboard.lua:109: in function <./lua/plugins/dashboard.lua:89>
 (tail call): ?
 [C]: in function 'xpcall'
 ./lua/main.lua:112: in function <./lua/main.lua:61>
 (tail call): ?
 polling.lua:47: in main chunk
 [C]: ?
it_IT
###cb:dashboard:settings:
```

**Comment by yangm97** [comment]
(Notice how lines changed because I removed some leftover code)

**Comment by yangm97** [comment]
Seems like /ban id is broken too

```
./lua/plugins/banhammer.lua:99: bad argument #2 to 'format' (string expected, got nil)
stack traceback:
 [C]: in function 'format'
 ./lua/plugins/banhammer.lua:99: in function <./lua/plugins/banhammer.lua:46>
 (tail call): ?
 [C]: in function 'xpcall'
 ./lua/main.lua:112: in function <./lua/main.lua:61>
 (tail call): ?
 polling.lua:47: in main chunk
 [C]: ?
en
/ban [numeric id]
```

**Comment by riskey95** [comment]
use /ban [user id] right?

**Comment by yangm97** [comment]
Maybe users are trying /ban with invalid numeric ids?

**Comment by yangm97** [comment]
Or rather, ids from members that are not in the chat.

**Comment by riskey95** [comment]
i dont get any error when tried to ban member that were not in the chat
![screenshot_20180716-112929](https://user-images.githubusercontent.com/3850438/42743362-45fdbad8-88ed-11e8-9326-bcf37a32056a.png)
![screenshot_20180716-112940](https://user-images.githubusercontent.com/3850438/42743363-466e3434-88ed-11e8-878d-88e6fa9d098a.png)

**Comment by yangm97** [comment]
@riskey95 So I don't have any idea what it could be then. We're planning to store problematic updates somewhere in the future to help debugging this kind of issues though.

**Comment by yangm97** [comment]
@NotAFile added an assert on `getnames_complete()` and today I just saw this:

```
An #error occurred.
./lua/utilities.lua:787: kicked is false/nil
stack traceback:
 [C]: in function 'assert'
 ./lua/utilities.lua:787: in function 'getnames_complete'
 ./lua/plugins/banhammer.lua:58: in function <./lua/plugins/banhammer.lua:46>
 (tail call): ?
 [C]: in function 'xpcall'
 ./lua/main.lua:122: in function 'on_msg_receive'
 ./lua/main.lua:335: in function 'parseMessageFunction'
 polling.lua:47: in main chunk
 [C]: ?
en
/ban [numeric id]
```

So there are cases where this function is returning nil/false.

**Comment by yangm97** [comment]
Is anyone against adding a default value, like “unknown” for this function?

**Comment by NotAFile** [comment]
@yangm97 sounds great

**Comment by yangm97** [comment]
Alright, should be fixed by fc12590 now.
